### PR TITLE
Fix some error with manifests

### PR DESCRIPTION
**Pull request by jeremco**
Hi,

This PR fixes some pb using segment templates manifest :  

Using manifest  https://dash.akamaized.net/akamai/bbb_30fps/bbb_30fps.mpd
- BaseURL of segment was not well formed (././.... instead of http://dash.akamaized.net/akamai/bbb_30fps/...)
- Incorrect number of media segments
- Error in SegmentTemplate computing, segment were starting at index 2, instead of index 1

Using manifest https://vm2.dashif.org/dash/vod/testpic_2s/multi_subs.mpd : 
- Incorrect number of mediaUrl objects

To test these manifests, in example, I have added process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0"; to ignore self-certificate errors. This is only set in electron sample, not in library

Regards
Jeremie

### CODE REVIEW

File: examples/load-manifests-examples.js [hunk 1]
```diff
@@ -9,7 +9,9 @@ function addExamples () {
   const examples = [
     '-',
     'http://demo.unified-streaming.com/video/ateam/ateam.ism/ateam.mpd',
-    'http://storage.googleapis.com/shaka-demo-assets/angel-one/dash.mpd'
+    'http://storage.googleapis.com/shaka-demo-assets/angel-one/dash.mpd',
+    'https://dash.edgesuite.net/akamai/bbb_30fps/bbb_30fps.mpd',
```
Comment (obecny) [inline examples/load-manifests-examples.js]: I think in repo there should be only stable examples and if served via https the certificate shouldn't be self signed. I think we can create a separate file, something like "manifests_examples.md" where we can add more edge cases with some explanations what it does and how and when to use it.

File: index.js [hunk 2]
```diff
@@ -5,6 +5,8 @@ const downstreamElectron = require('./api/index');
 // TESTING PRODUCTION
 // const downstreamElectron = require('./dist/index');
 
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
```
Comment (obecny) [inline index.js]: I guess this is only needed if you want to use servers with self signed certificate, but there should be some explanation here and I would rather move it to `readme.md`

File: api/downloads/download-file.js [hunk 3]
```diff
@@ -160,10 +160,10 @@ DownloadFile.prototype._onDownloadFailure = function (err, aborted) {
     if (this._errors <= this._options.maxDownloadRetry) {
       this._retryDownload();
     } else {
-      this.emit("error", err);
+      this.emit("error", {message: err});
```
Comment (obecny) [inline api/downloads/download-file.js]: can you explain what it does and why is this change necessary ?

File: api/manifest/parser/segment-information.js [hunk 4]
```diff
@@ -163,6 +163,10 @@ const SegmentInformation = (function () {
     const segmentDuration = parseInt(this.segmentTemplate.attributes.getNamedItem("duration").nodeValue);
     const segmentTimescale = (this.segmentTemplate.attributes.getNamedItem("timescale")) ? parseInt(
         this.segmentTemplate.attributes.getNamedItem("timescale").nodeValue) : 1;
+
+    // use math.ceil, instead of math.floor, because floor returns the largest integer less than or equal to a given number.
+    // math.ceil returns the smallest integer greater than or equal to a given number.
+    // using math.floor may lead to an incorrect number of segments.
```
Comment (obecny) [inline api/manifest/parser/segment-information.js]: I think tracking previous errors with comments might not be needed, would rather have a unit test for that if you think it is important, do you agree ?

File: api/downloads/download-file.js (hunk 3 above)
Comment (jeremco) [reply api/downloads/download-file.js]: Hi 
That is because onError callback is declared as : 
Download.prototype._onError = function (data) {
  const self = this;
  this.status = STATUSES.ERROR;
  data = data || {};
  const message = data.message || "";

  self._removeEvents();
  self._updateStats();
  self.events.emit("error", self, message);
};

Message is important to display or log the error.
But maybe it can be treated in another PR

File: api/manifest/parser/segment-information.js (hunk 4 above)
Comment (jeremco) [reply api/manifest/parser/segment-information.js]: The comment was added before you make the change in master, to explain why ceil has to  be called instead of floor.
It can be removed

File: examples/load-manifests-examples.js (hunk 1 above)
Comment (jeremco) [reply examples/load-manifests-examples.js]: I do not agree. The sample is the only way you can check there is no regression in the library. And each time a pb appears on a stream, we must add it to the sample, so that, on next delivery, we'll be able to test it without having to modify the sample code.

Moreover, how could you test the corretion of the pr if the stream is not in the sample ?

File: index.js (hunk 2 above)
Comment (jeremco) [reply index.js]: Your are right, a comment would be welcome

**Comment by obecny** [comment]
closing this as this is not relevant anymore, thx
